src/proc_cpuview.c

```diff
@@ -937,6 +966,28 @@ int proc_cpuinfo_read(char *buf, size_t size, off_t offset,
 		return total_len;
 	}
 
+	if (get_host_personality(&host_personality) < 0)
```

Comment:
I don't think we should call this here. Instead we should call this once when the shared module is loaded in the function marked as constructor in `bindings.c`.

So something like the following (UNTESTED):

```diff
diff --git a/src/bindings.c b/src/bindings.c
index 422a622..5ff6a76 100644
--- a/src/bindings.c
+++ b/src/bindings.c
@@ -45,9 +45,15 @@ static bool can_use_swap;
 static bool can_use_sys_cpu;
 static bool has_versioned_opts;
 static bool memory_is_cgroupv2;
+static __u32 host_personality;

 static volatile sig_atomic_t reload_successful;

+__u32 liblxcfs_personality(void)
+{
+       return host_personality;
+}
+
 bool liblxcfs_functional(void)
 {
        return reload_successful != 0;
@@ -919,6 +925,11 @@ static void __attribute__((constructor)) lxcfs_init(void)
                goto broken_upgrade;
        }

+       if (get_host_personality(&host_personality) < 0) {
+               lxcfs_info("Failed to retrieve host personality");
+               goto broken_upgrade;
+       }
+
        reload_successful = 1;
        return;

diff --git a/src/bindings.h b/src/bindings.h
index e35423a..1624784 100644
--- a/src/bindings.h
+++ b/src/bindings.h
@@ -107,6 +107,7 @@ extern bool liblxcfs_can_use_swap(void);
 extern bool liblxcfs_memory_is_cgroupv2(void);
 extern bool liblxcfs_can_use_sys_cpu(void);
 extern bool liblxcfs_has_versioned_opts(void);
+extern __u32 liblxcfs_personality(void);

 static inline bool lxcfs_has_opt(struct lxcfs_opts *opts, lxcfs_opt_t opt)
 {
```

Reply:
Thanks! Have done. :-)